extension/storage/filestorage/extension.go

```diff
@@ -59,9 +57,6 @@ func (lfs *localFileStorage) Start(context.Context, component.Host) error {
 
 // Shutdown will close any open databases
 func (lfs *localFileStorage) Shutdown(context.Context) error {
-	for _, client := range lfs.clients {
-		client.close()
-	}
```

Comment:
It is not clear how we can do garbage collection after this change. As TODO says we were planning to delete everything that does not have a client. Now that we no longer track clients how do we do that? TODO is now irrelevant/impossible.

Reply:
Ah, good point. I will remove the comment. 

In the future, I think a mechanism very similar to the previous design should work for cleanup. Each time `GetClient` is called, we can capture the filename associated with the client that is created. When the extension shuts down, we can use the same logic that we had intended.

Reply:
Every component that gets a client should close it when they are shutdown, so not sure if this extension should do the cleanup.